### PR TITLE
Added ProjectIdUtilTest.java and GiteeClientStub.java, fixed edge cases

### PR DESCRIPTION
**Pull request by Puffy1215**
Added `ProjectIdUtilTest.java` and `GiteeClientStub.java`, fixed edge cases with `:` characters and `.com` in `projectId` .

### Testing done
Ran and compiled on Ubuntu 24. All test cases passed.

### Submitter checklist
- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [x] Link to relevant issues in GitHub or Jira
- [x] Link to relevant pull requests, esp. upstream and downstream changes
- [x] Ensure you have provided tests that demonstrate the feature works or the issue is fixed

### CODE REVIEW

File: src/main/java/com/gitee/jenkins/util/ProjectIdUtil.java [hunk 1]
```diff
@@ -23,6 +23,11 @@ public static String retrieveProjectId(GiteeClient client, String remoteUrl) thr
             String projectId;
             if (baseUri != null && remoteUrl.startsWith(baseUri)) {
                 projectId = new URIish(remoteUrl.substring(baseUri.length())).getPath();
+                if (projectId.contains(":")) {
+                    projectId = projectId.substring(projectId.indexOf(":"));
```
Comment (MarkEWaite) [inline src/main/java/com/gitee/jenkins/util/ProjectIdUtil.java:27]: When I ran this from the Groovy script console inside Jenkins, it returned a string that starts with ':'.  Is that expected?  I assumed you want the resulting string to not include the ':' character
```suggestion
                    projectId = projectId.substring(projectId.indexOf(":") + 1);
```

I was also surprised at the removal of the preceding characters, since that seems like it would not identify the project ID correctly.

However, I don't understand Gitee well enough to be sure of either of those things and I can't read Chinese at all.  That means you are welcome to ignore my review, since it is coming from my guesses, not from anything that I've investigated.

File: src/main/java/com/gitee/jenkins/util/ProjectIdUtil.java [hunk 2]
```diff
@@ -23,6 +23,11 @@ public static String retrieveProjectId(GiteeClient client, String remoteUrl) thr
...
+                } else if (projectId.startsWith(".")) {
+                    projectId = projectId.substring(projectId.indexOf("/"));
```
Comment (MarkEWaite) [inline src/main/java/com/gitee/jenkins/util/ProjectIdUtil.java:29]: Is it always assured that there will be a '/' character in any projectId that starts with '.'?  My first interactive test with the Groovy script console used a string that included '.' but did not include '/'.  That threw an exception.

**Review by MarkEWaite — COMMENTED** [review]
Optional questions and comments for your consideration.
